include/alpaka/mem/buf/BufOacc.hpp

```diff
@@ -0,0 +1,650 @@
+            typename TElem,
+            typename TDim,
+            typename TIdx>
+        struct GetPitchBytes<
+            TIdxIntegralConst,
+            BufOacc<TElem, TDim, TIdx>>
+        {
+            //-----------------------------------------------------------------------------
+            ALPAKA_FN_HOST static auto getPitchBytes(
+                BufOacc<TElem, TDim, TIdx> const & pitch)
+            -> TIdx
+            {
+                return (*pitch).m_pitchBytes[TIdxIntegralConst::value];
+            }
+        };
+
+        //#############################################################################
+        //! The BufOacc 1D memory allocation trait specialization.
+        template<
+            typename TElem,
+            typename TIdx>
+        struct BufAlloc<
+            TElem,
+            DimInt<1u>,
+            TIdx,
+            DevOacc>
+        {
+            //-----------------------------------------------------------------------------
+            template<
+                typename TExtent>
+            ALPAKA_FN_HOST static auto allocBuf(
+                DevOacc const & dev,
+                TExtent const & extent)
+            -> BufOacc<TElem, DimInt<1u>, TIdx>
+            {
+                ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
+
+                auto const width(extent::getWidth(extent));
+                auto const widthBytes(width * static_cast<TIdx>(sizeof(TElem)));
+
+                dev.makeCurrent();
+                void * memPtr = acc_malloc(static_cast<std::size_t>(widthBytes));
+
+#if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                std::cout << __func__
+                    << " ew: " << width
+                    << " ewb: " << widthBytes
+                    << " ptr: " << memPtr
+                    << " device: " << dev.m_spDevOaccImpl->iDevice()
+                    << std::endl;
+#endif
+                return
+                    BufOacc<TElem, DimInt<1u>, TIdx>(
+                        dev,
+                        reinterpret_cast<TElem *>(memPtr),
+                        extent);
+            }
+        };
+
+        //#############################################################################
+        //! The BufOacc nD memory allocation trait specialization. \todo Add pitch
+        template<
+            typename TElem,
+            typename TDim,
+            typename TIdx>
+        struct BufAlloc<
+            TElem,
+            TDim,
+            TIdx,
+            DevOacc>
+        {
+            //-----------------------------------------------------------------------------
+            template<
+                typename TExtent>
+            ALPAKA_FN_HOST static auto allocBuf(
+                DevOacc const & dev,
+                TExtent const & extent)
+            -> BufOacc<TElem, TDim, TIdx>
+            {
+                ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
+
+                const std::size_t size = static_cast<std::size_t>(extent.prod()) * sizeof(TElem);
```

Comment:
This should probably be:
```suggestion
                const std::size_t size = static_cast<std::size_t>(extent::getExtentVec(extent).prod()) * sizeof(TElem);
```
Else we can not be sure that we can call `.prod` on it.

Reply:
Could you apply this to the OMP implementation as well please?

Reply:
About the OMP5: do you mean in this PR or in a separate one?

Reply:
I do not care. Can be this one as well.

Reply:
Included.